src/argilla/client/feedback/dataset/base.py

```diff
@@ -298,19 +211,44 @@ def question_by_name(self, name: str) -> AllowedQuestionTypes:
             f" {', '.join(q.name for q in self._questions)}"
         )
 
-    @property
-    def records(self) -> List[FeedbackRecord]:
-        """Returns the all the records in the dataset."""
-        return self._records + self._new_records
+    def _validate_records(
+        self, records: Union[FeedbackRecord, Dict[str, Any], List[Union[FeedbackRecord, Dict[str, Any]]]]
+    ) -> List[FeedbackRecord]:
+        if isinstance(records, list):
+            if len(records) == 0:
+                raise ValueError("Expected `records` to be a non-empty list of `dict` or `FeedbackRecord`.")
+            new_records = []
+            for record in records:
+                if isinstance(record, dict):
+                    new_records.append(FeedbackRecord(**record))
+                elif isinstance(record, FeedbackRecord):
+                    new_records.append(record)
+                else:
+                    raise ValueError(
+                        f"Expected `records` to be a list of `dict` or `FeedbackRecord`, got type {type(record)} instead."
+                    )
+            records = new_records
+        elif isinstance(records, dict):
+            records = [FeedbackRecord(**records)]
+        elif isinstance(records, FeedbackRecord):
+            records = [records]
+        else:
+            raise ValueError(
+                f"Expected `records` to be a `dict` or `FeedbackRecord`, got type {type(records)} instead."
+            )
 
-    def iter(self, batch_size: Optional[int] = FETCHING_BATCH_SIZE) -> Iterator[List[FeedbackRecord]]:
-        """Returns an iterator over the records in the dataset.
+        if self._fields_schema is None:
+            self._fields_schema = generate_pydantic_schema(self.fields)
```

Comment:
Does it mean that once the pydantic schema is generated the field_schema is not changed after a field setting update? Maybe we can just generate each time the method is called:
```suggestion
        fields_schema  = generate_pydantic_schema(self.fields)
```
Ow, are we using the self._fields_schema outside this method?

Reply:
No we're not, but ideally we want to re-use it, instead of calculating it every time, and since we don't support field settings updates for the moment I'd say we can leave it as is